### ISSUE TITLE
Temperature Change Error 

### ISSUE THREAD

**Issue by bstempien** [PeteRager/lennoxs30#154]
Upfront - I am brand spanking new to Home Assistant, so likely to be some kind of issue on my part. 

I followed the instructions to get HACS installed and your Lennox iComfort Component on a fresh HAOS 7.6. 
I have configured for local connection to my single iComfort S30 Thermostat. 

I can see the Device in the Overview screen, but if bring up the Details screen with the 3 dot menu and try to change my upper or lower temp, it fails. I am in Heat/Cool mode. In the terminal I can find the following log messages in the homeassistant.log. 

2022-05-16 17:44:39 ERROR (MainThread) [homeassistant.helpers.script.websocket_api_script] websocket_api script: Error executing script. Invalid data for call_service at pos 1: some but not all values in the same group of inclusion 'temperature' @ data[<temperature>]
2022-05-16 17:44:39 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [139781575788288] Error handling message: some but not all values in the same group of inclusion 'temperature' @ data[<temperature>]. Got None (invalid_format)

**Comment by PeteRager** [comment]
Thanks for the disclosure 😀. 

Let’s try some things one at a time, to see what is working and what is not.  Those errors appear to be before the integration has had a chance to handle it.

There are two ways the Lennox Thermostat works in Heat/Cool mode.  One mode uses two setpoints and the other uses one (called perfect mode).  Which mode are you in?

Try these tests from your dashboard.   

1.  Zone Mode control.  Can you change the mode of the zone?  From the thermostat card, try pressing Off button, Cool button, Heat button, Heat/Cool button.
2. Place the zone in Heat Mode and try to adjust the temperature - did it work?
3. Place the zone in Cool Mode and try to adjust the temperature - did it work?
4. Place the zone back in Heat/Cool mode and try to adjust the temperature (this should fail)

**Comment by bstempien** [comment]
Ran thru the tests. And it did everything correctly. Including now
switching the High temp when in heat/cool mode.

Maybe since things were fairly new something had not fully registered until
I went through all the settings? I got the iComfort setup last night, so
there was several hours between setup and the first change I tried to make.

Either way, seems to be working now. I was even able to figure out the
Automations and determined to use a Service to setup the temp changes, vs.
device. Pretty steep learning curve compared to SmartThings, but looking
much more promising.

Thank you for your hard work on this component. It is the sole reason I
decided to start down the Home Assistant path away from Samsung.
